RELEASE.md

```diff
@@ -1,3 +1,25 @@
+# Releases
+
+## The `master` branch
+
+The `master` branch is where the active Kuma development happens. This is where all the new features, critical and minor fixes get accepted. Branch from this one when starting to work on a new PR.
+
+## Release branches
+
+The release branches are named in the format `release-A.B`, where `A` is the major release number and `B` is the minor release number. These branches are considered stable and hold the tagged releases of Kuma. The tags are in the format `A.B.C` where `A` is the major, `B` is the minor and `C` is the patch level of the released version.
+
+### Releases
+
+The process to release a Kuma version is as follows:
+ * Create the release branch `release-A.B` and tag `A.B.0-rc1`. This also includes the updated GUI for the release.
+ * Next 2 days, test the `rc1` version for the backward compatibility, breaking changes and upgrade instruction. Verify the newly added feature are working as expected. Check the documentation reflects the breaking changes and the new functionality.
+ * If a critical problem is found, fix it and release `rc2`. Run the test/verification procedures again
+ * When there is confidence tag and release `A.B.0` following the checklist below
+
+### Patch releases
+
+After the initial release there may be a need for a patch release `A.B.1`, `A.B.2` and so on. These typically include only fixes to critical issues, but there might be exceptions where features are backported from master. I most cases the patch releases follow a lighter weight release procedure, without `rc` versions and a testing/verification cycle focused on the chanaged/fixed functionalities.  
```

Comment:
```suggestion
After the initial release there may be a need for a patch release `A.B.1`, `A.B.2` and so on. These typically include only fixes to critical issues, but there might be exceptions where features are backported from master. In most cases the patch releases follow a lighter weight release procedure, without `rc` versions and a testing/verification cycle focused on the changed/fixed functionalities.  
```

Reply:
@austince can you please make a PR against `master` too?